pkg/aws/policies.go

```diff
@@ -448,6 +493,102 @@ func (c *awsClient) HasCompatibleVersionTags(iamTags []*iam.Tag, version string)
+		if aws.StringValue(tag.Key) == tags.OpenShiftVersion {
+			if version == aws.StringValue(tag.Value) {
+				return true, nil
+			}
+
+			upgradeVersion, err := semver.NewVersion(version)
+			if err != nil {
+				return false, err
+			}
+
+			currentVersion, err := semver.NewVersion(aws.StringValue(tag.Value))
+			if err != nil {
+				return false, err
+			}
+
+			upgradeVersionSegments := upgradeVersion.Segments64()
+			c, err := semver.NewConstraint(fmt.Sprintf(">= %d.%d",
+				upgradeVersionSegments[0], upgradeVersionSegments[1]))
+			if err != nil {
+				return false, err
+			}
+			return c.Check(currentVersion), nil
+		}
+	}
+	return false, nil
+}
+
+// ValidateClusterAccountRolesVersion ensures all account roles for a cluster is compatible
+// with a specific openshift version
+func (c *awsClient) ValidateClusterAccountRolesVersion(cluster *cmv1.Cluster, version string, creator *Creator) (
+	bool, error) {
+	var roleNames []string
+	instanceRoles, hasInstanceRoles := cluster.AWS().STS().GetInstanceIAMRoles()
+	if !hasInstanceRoles {
+		return false, errors.Errorf("Can not validate cluster roles, "+
+			"instance roles for cluster '%s' does not exist",
+			cluster.Name())
+	}
+
+	roleNames = append(roleNames, getRoleNameFromARNSplit(instanceRoles.MasterRoleARN()))
+	roleNames = append(roleNames, getRoleNameFromARNSplit(instanceRoles.WorkerRoleARN()))
+
+	installerRole, hasInstallerRole := cluster.AWS().STS().GetRoleARN()
+	if !hasInstallerRole {
+		return false, errors.Errorf("Can not validate cluster roles, role arn for cluster '%s' does not exist",
+			cluster.Name())
+	}
+	roleNames = append(roleNames, getRoleNameFromARNSplit(installerRole))
+
+	supportRole, hasSupportRole := cluster.AWS().STS().GetSupportRoleARN()
+	if !hasSupportRole {
+		return false, errors.Errorf("Can not validate cluster roles, "+
+			"support role arn for cluster '%s' does not exist",
+			cluster.Name())
+	}
+	roleNames = append(roleNames, getRoleNameFromARNSplit(supportRole))
+
+	for _, role := range roleNames {
+		isCompatible, err := c.validateRoleUpgradeVersionCompatibility(role, version, creator.AccountID)
+		if err != nil {
+			return false, errors.Errorf("Error checking role '%s' compatibility : %v", role, err)
+		}
+		if !isCompatible {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+// ValidateClusterOperatorRolesVersion ensures all operator roles for a cluster is compatible with a specific
+// openshift version
+func (c *awsClient) ValidateClusterOperatorRolesVersion(cluster *cmv1.Cluster, version string, creator *Creator) (
+	bool, error) {
+	operatorRoles, hasOperatorRoles := cluster.AWS().STS().GetOperatorIAMRoles()
+	if !hasOperatorRoles {
+		return false, errors.Errorf("Can not validate cluster roles, operator roles for cluster '%s' does not exist",
+			cluster.Name())
+	}
+	for _, role := range operatorRoles {
+		isCompatible, err := c.validateRoleUpgradeVersionCompatibility(getRoleNameFromARNSplit(role.RoleARN()),
```

Comment:
I mentioned in other PR as well we cannot use the same method for operator role and account role